src/anyio/to_interpreter.py

```diff
@@ -0,0 +1,208 @@
+from __future__ import annotations
+
+import atexit
+import os
+import pickle
+import sys
+from collections import deque
+from collections.abc import Callable, Mapping
+from textwrap import dedent
+from typing import Any, TypeVar
+
+from . import to_thread
+from ._core._exceptions import BrokenWorkerIntepreter
+from ._core._synchronization import CapacityLimiter
+from .lowlevel import RunVar
+
+if sys.version_info >= (3, 11):
+    from typing import TypeVarTuple, Unpack
+else:
+    from typing_extensions import TypeVarTuple, Unpack
+
+UNBOUND = 2  # I have no clue how this works, but it was used in the stdlib
+FMT_UNPICKLED = 0
+FMT_PICKLED = 1
+DEFAULT_CPU_COUNT = 8  # this is just an arbitrarily selected value
+
+T_Retval = TypeVar("T_Retval")
+PosArgsT = TypeVarTuple("PosArgsT")
+
+_idle_workers = RunVar[deque["Worker"]]("_available_workers")
+_default_interpreter_limiter = RunVar[CapacityLimiter]("_default_interpreter_limiter")
+
+
+class Worker:
+    _run_func = compile(
+        dedent("""
+        import _interpqueues as queues
+        import _interpreters as interpreters
+        from pickle import loads, dumps, HIGHEST_PROTOCOL
+
+        item = queues.get(queue_id)[0]
+        try:
+            func, args, kwargs = loads(item)
+            retval = func(*args, **kwargs)
+        except Exception as exc:
+            is_exception = True
+            retval = exc
+        else:
+            is_exception = False
+
+        try:
+            queues.put(queue_id, (retval, is_exception), FMT_UNPICKLED, UNBOUND)
+        except interpreters.NotShareableError:
+            retval = dumps(retval, HIGHEST_PROTOCOL)
+            queues.put(queue_id, (retval, is_exception), FMT_PICKLED, UNBOUND)
+        """),
+        "<string>",
+        "exec",
+    )
+
+    _interpreter_id: int
+    _queue_id: int
+
+    async def initialize(self) -> None:
```

Comment:
```suggestion
    def initialize(self) -> None:
```

does this need to be async?

Reply:
No, and in fact it should not set up the interpreter in the event loop thread either. I'll fix both issues at once.